pkg/crosscluster/producer/replication_manager.go

```diff
@@ -450,26 +450,42 @@ func (r *replicationStreamManagerImpl) AuthorizeViaJob(
 	return nil
 }
 
-func (r *replicationStreamManagerImpl) AuthorizeViaReplicationPriv(ctx context.Context) error {
-	err := r.evalCtx.SessionAccessor.CheckPrivilege(ctx,
+// AuthorizeViaReplicationPriv ensures the user has the REPLICATIONSOUCE privilege. If tableNames is passed, then table level auth is tried.
+func (r *replicationStreamManagerImpl) AuthorizeViaReplicationPriv(
+	ctx context.Context, tableNames ...string,
+) (err error) {
+	defer func() {
+		if err == nil {
+			r.authorized = true
+		}
+	}()
+
+	err = r.evalCtx.SessionAccessor.CheckPrivilege(ctx,
```

Comment:
Is explicitly checking the global and table permission required here? I would expect the lower level privilege infrastructure to do this for us. Looking at some other  callers, `AuthorizeChangeFeed` job appears to only check the table level permissions.